src/netbase.cpp

```diff
@@ -572,6 +710,21 @@ bool GetProxy(enum Network net, proxyType &proxyInfoOut) {
     return true;
 }
 
+/**
+ * Set the name proxy to use for all connections to nodes specified by a
+ * hostname. After setting this proxy, connecting to a node sepcified by a
+ * hostname won't result in a local lookup of said hostname, rather, connect to
+ * the node by asking the name proxy for a proxy connection to the hostname,
+ * effectively delegating the hostname lookup to the specified proxy.
+ *
+ * This delegation increases privacy for those who set the name proxy as they no
+ * longer leak their external hostname queries to their DNS servers.
+ *
+ * @returns Whether or not the operation succeeded.
+ *
+ * @note Alternatively, SOCKS5 does have support for UDP-over-SOCKS5, but a DNS
+ *       resolver is beyond the scope of this project.
```

Comment:
Note that no SOCK5 server in common use, most notably Tor, actually implements UDP support. So it'd be kind of pointless to do this.
The [Tor socks extension spec](https://github.com/torproject/torspec/blob/master/socks-extensions.txt) does mention a RESOLVE command that they implement that is specific to Tor, this would be a more realistic option.

Reply:
@laanwj Cool, will include. I'm thinking we use `@note` as a place to discuss the design space we've considered and why they don't work so it's recorded somewhere for future developers. Does that good to you? We could also do this somewhere else and reference from `@note`.

Reply:
Yes, sounds good to me, I was just mentioning it to prevent people from going up this road to implement this (reading it as a TODO) only to conclude it's pointless, so it makes sense to mention that.

Reply:
Reworded a little bit to make it more clear this is just a design space comment not a TODO,